bot/cogs/starboard.py

```diff
@@ -204,29 +203,52 @@ def check(r, u):
         await original_msg.clear_reactions()
         log.info(f"Cleared all reactions of message {original_id}")
 
-        await ctx.message.add_reaction(OK_HAND)
+        await ctx.message.add_reaction(Emojis.ok_hand)
         await asyncio.sleep(3)
         await ctx.message.delete()
 
-    async def on_raw_reaction_remove(self, payload: RawReactionActionEvent):
+    def failed_these_checks(self, payload: RawReactionActionEvent):
         if payload.guild_id != Guild.id:
-            return log.debug("Reaction was not added in the correct guild.")
+            log.debug("Reaction was not added in the correct guild.")
+            return True
+
+        if payload.emoji.name != Emojis.lvl1_star:
+            log.debug("Invalid emoji was reacted")
+            return True
+
+        if payload.channel_id == Channels.starboard:
+            log.debug("Can't star a message on the starboard.")
+            return True
+
+        guild = self.bot.get_guild(payload.guild_id)
+        starrer_member = guild.get_member(payload.user_id)
+
+        if not any(role.id in Filter.allowed_to_star for role in starrer_member.roles):
+            log.debug("Member was not allowed to star a message")
```

Comment:
I really think we're going overboard with all this debug logging. I'd much prefer trace logging. These are all very straightforward if statements.

Reply:
I have not figured out what those are meant for, yet.  Please do educate me.

Reply:
`log.debug` -> `log.trace`